### PR TITLE
Integrate DIF Resolver

### PR DESCRIPTION
**Pull request by Exulansis**
Partially addresses #326 

Integrates the DID-Resolver module into our library. Reduces the usage of the `IRegistry` interface, offloading some of the responsibility /complexity to the `jolo-did-method` module.

The API on functions making use of a resolver (e.g. `utils/validation`, `identityWallet.validateJWT`) has been updated to no longer take a `IRegistry` instance, but rather take a `Registry` class instead. This is a **breaking** API change.

### CODE REVIEW

File: tests/integration/authentication.integration.ts [hunk 1]
```diff
@@ -16,34 +16,37 @@ const expect = chai.expect
...
   })
 
   it('Should allow for consumption of valid authentication request token by user', async () => {
     const decodedAuthRequest = JSONWebToken.decode<Authentication>(
-      authRequestEncoded,
+      authRequestJWT.encode(),
     )
-    expect(decodedAuthRequest.interactionToken).to.be.instanceOf(Authentication)
+
+    // expect(decodedAuthRequest.interactionToken).to.be.instanceOf(Authentication)
```
Comment (chunningham) [inline tests/integration/authentication.integration.ts:43]: each of these `expect().to.be.instanceOf()`s can be uncommented once #404 is in the trunk, just a note for when merging happens

**Comment by mnzaki** [comment]
IMHO change this PR to the `next` branch and break the `IRegistry` interface

File: tests/integration/authentication.integration.ts (hunk 1 above)
Comment (Exulansis) [reply tests/integration/authentication.integration.ts]: Done.

**Comment by Exulansis** [comment]
> IMHO change this PR to the `next` branch and break the `IRegistry` interface

+1 The next branch is slightly out of date atm.

File: tests/integration/payment.integration.ts [hunk 2]
```diff
@@ -27,6 +27,10 @@ describe('Integration Test - EXPERIMENTAL Token interaction flow Payment', () =>
       description: 'Payment for monthly subscription to awesome service',
       transactionOptions: {
         value: 0.5e18, // 0.5 Ether in Wei
+        // TODO DEFAULT VALUES NOT POPULATED YET DUE TO #399
```
[the remark this answers is not in the document]
Comment (Exulansis) [reply tests/integration/payment.integration.ts:30]: Addressed in #406

**Comment by Exulansis** [comment]
I've changed the base to `next`.
It's out of date, and has a few PRs merged into it which `develop` does not contain.

We should decide what to do with `next` to proceed. I'd propose resetting it to match current developm or merge develop in, if the PRs there are useful.

**Comment by Exulansis** [comment]
Develop should be merged into `next`.

File: ts/registries/jolocomRegistry.ts [hunk 3]
```diff
@@ -283,6 +265,13 @@ export const createJolocomRegistry = (
   jolocomRegistry.ethereumConnector = ethereumConnector
   jolocomRegistry.contractsAdapter = contracts.adapter
   jolocomRegistry.contractsGateway = contracts.gateway
+  jolocomRegistry.resolver = jolocomResolver()
 
   return jolocomRegistry
 }
+
+export const jolocomResolver = (additionalResolver = {}): Resolver => {
```
Comment (chunningham) [inline ts/registries/jolocomRegistry.ts:273]: Could we consider removing the resolver class entirely and only use the did-resolver?

Comment (Exulansis) [reply ts/registries/jolocomRegistry.ts:273]: The `Resolver` class here is the `did-resolver`

Comment (chunningham) [reply ts/registries/jolocomRegistry.ts:273]: oh :sweat_smile:

**Pull request merged**